bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -209,7 +209,9 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
 
         thingBuilder.withConfiguration(configuration)
         thingBuilder.withBridge(bridgeUID)
-        thingBuilder.withLabel(label)
+        if(label !== null) {
```

Comment:
Why is location nullable, while label isn't?
Shouldn't we treat the Builders as consumer API and thus accept null values as parameter unless it causes real issues (as e.g. in the create methods)?